secure-enclave/values.yaml

```diff
@@ -2,42 +2,100 @@
-imagePullSecrets: []
 # This is to override the chart name.
 nameOverride: ""
 fullnameOverride: ""
 
+## @param aws this defines the values configured when deployed on AWS.
+##
 aws:
-  enabled: true
+  ## @param aws.enabled this defines if AWS configurations are enabled.
+  ##
+  enabled: false
+  ## @param aws.storageClass this defines the storage class used for the deployment
+  ##
   storageClass:
+    ## @param aws.storageClass.name this defines the name of the storage class
+    ##
     name: aws-ebs-sc
+    ## @param aws.storageClass.provisioner this defines the storage class provisioner 
+    ##
     provisioner: ebs.csi.aws.com
+    ## @param aws.storageClass.volumeBindingMode this defines the storage class volume binding mode 
+    ##
     volumeBindingMode: WaitForFirstConsumer
+    ## @param aws.storageClass.parameters this defines the storage class parameters 
+    ##
     parameters:
+      ## @param aws.storageClass.parameters.csi.storage.k8s.io/fstype this defines the filesystem type for the storage class 
+      ##
       csi.storage.k8s.io/fstype: xfs
+      ## @param aws.storageClass.parameters.type this defines the type of the storage class 
+      ##
       type: io1
+      ## @param aws.storageClass.parameters.iopsPerGB this defines the IOPS per GB for the storage 
+      ##
       iopsPerGB: "50"
+      ## @param aws.storageClass.parameters.encrypted  this defines if the storage is encrypted 
+      ##
       encrypted: "true"
+    ## @params aws.storageClass.allowedTopologies this defines the allowed toplogies, for eg. the zone where the storage class will be created in.
+    ## 
     allowedTopologies:
       - matchLabelExpressions:
           - key: topology.ebs.csi.aws.com/zone
             values:
               - us-east-1
 
+## @param networkPolicy this defines if the network policy restrictions are enforced.
+## Few limitations: 
+## FQDNs Egress restrictions
+## TODO. Enforce Ingress only to some pods
+##
+networkPolicy:
+  ## @param networkPolicy.enabled this enables or disables the network policy
+  enabled: false
+
+## @param managementApp this defines all the configurations related to the management app
+##
+managementApp: 
+  ## @param managementApp.memberId this defines the id of the member deploying the enclave
+  ##
+  memberId:
+  ## @param managementApp.endpoint this defines the endpoint where the management app is available. 
+  ##
+  endpoint: 
+    ## @param managementApp.endpoint.protocol this defines the protocol used to access the management app 
+    ##
+    protocol: https
+    ## @param managementApp.endpoint.host this defines the host where the management app is available
+    ## This value is required
+    host: 
+    ## @param managementApp.endpoint.port this defines the port where the management app is available
+    ##
+    port: 443
+  ## @param managementApp.privateKey this defines the private key of the member deploying the enclave
+  ##
+  privateKey: |
```

Comment:
Best practice for values files is to check them into GitHub, and they cannot do that with their private key. Refactor this so it references a k8s secret, and update the docs to require the member to create the secret before the helm deploy.